diff --git a/make/gensrc/Gensrc-java.base.gmk b/make/gensrc/Gensrc-java.base.gmk
--- a/make/gensrc/Gensrc-java.base.gmk
+++ b/make/gensrc/Gensrc-java.base.gmk
@@ -33,7 +33,7 @@
 include GensrcCharsetCoder.gmk
 include GensrcBuffer.gmk
 include GensrcExceptions.gmk
-include GensrcVarHandles.gmk
+#include GensrcVarHandles.gmk
 include GensrcModuleLoaderMap.gmk

 ################################################################################

