diff --git a/src/java.base/share/native/include/jvm.h b/src/java.base/share/native/include/jvm.h
--- a/src/java.base/share/native/include/jvm.h
+++ b/src/java.base/share/native/include/jvm.h
@@ -116,6 +116,16 @@


 /*
+ * java.util.zip.ZipFile                                                        //IBM-use_j9
+ */                                                                             //IBM-use_j9
+#define JVM_ZIP_HOOK_STATE_OPEN 1                                               //IBM-use_j9
+#define JVM_ZIP_HOOK_STATE_CLOSED 2                                             //IBM-use_j9
+                                                                                //IBM-use_j9
+JNIEXPORT void JNICALL                                                          //IBM-use_j9
+JVM_ZipHook(JNIEnv *env, const char* filename, jint newState);                  //IBM-use_j9
+                                                                                //IBM-use_j9
+                                                                                //IBM-use_j9
+/*                                                                              //IBM-use_j9
  * java.lang.Runtime
  */
 JNIEXPORT void JNICALL

