diff --git a/make/BuildNashorn.gmk b/make/BuildNashorn.gmk
--- a/make/BuildNashorn.gmk
+++ b/make/BuildNashorn.gmk
@@ -76,9 +76,9 @@
 ifeq ($(BOOT_JDK_MODULAR), true)
   NASGEN_OPTIONS := \
       -cp $(BUILDTOOLS_OUTPUTDIR)/nasgen_classes \
-      --patch-module java.base=$(BUILDTOOLS_OUTPUTDIR)/nasgen_classes \
-      --add-exports java.base/jdk.internal.org.objectweb.asm=ALL-UNNAMED \
-      --add-exports java.base/jdk.internal.org.objectweb.asm.util=ALL-UNNAMED \
+      -Xpatch:java.base=$(BUILDTOOLS_OUTPUTDIR)/nasgen_classes \
+      -XaddExports:java.base/jdk.internal.org.objectweb.asm=ALL-UNNAMED,java.base/jdk.internal.org.objectweb.asm.util=ALL-UNNAMED \
+      #--add-exports java.base/jdk.internal.org.objectweb.asm.util=ALL-UNNAMED \
       #
 else
   NASGEN_OPTIONS := \
